### ISSUE TITLE
Navigator has no active routes to replace.

### ISSUE THREAD

**Issue by niksen75** [flutter/flutter#125890]
```
════════ Exception caught by scheduler library ═════════════════════════════════
The following assertion was thrown during a scheduler callback:
Navigator has no active routes to replace.
'package:flutter/src/widgets/navigator.dart':
navigator.dart:1
Failed assertion: line 4647 pos 12: '_history.any(_RouteEntry.isPresentPredicate)'

Either the assertion indicates an error in the framework itself, or we should provide substantially more information in this error message to help you determine and fix the underlying cause.

**In either case, please report this assertion by filing a bug on GitHub:**
  https://github.com/flutter/flutter/issues/new?template=2_bug.md
  
 When the exception was thrown, this was the stack
#2      NavigatorState._pushReplacementEntry
navigator.dart:4647
#3      NavigatorState.pushReplacement
navigator.dart:4603
#4      NavigatorState.pushReplacementNamed
navigator.dart:4267
#5      Navigator.pushReplacementNamed
navigator.dart:1799
#6      _MyWidgetState.onWidgetBuilt
app_router.dart:226
#7      _MyWidgetState.didChangeDependencies.<anonymous closure>
```
---

```dart
class _MyWidgetState  extends State<MyWidget> {

  void onWidgetBuilt() {
    Navigator.pushReplacementNamed(context, '/some-route-name');
  }

  @override
  void didChangeDependencies() {
      SchedulerBinding.instance.addPostFrameCallback((duration) {
        onWidgetBuilt();
    });
  }
}
```

**Comment by danagbemava-nc** [comment]
Hi @niksen75,

Please provide your `flutter doctor -v` and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) along with steps to reproduce so that we can investigate this issue. 

Thank you

**Comment by niksen75** [comment]
I'm listening to firebaseAuth.userChanges().listen((onData) {...}) and updating a ChangeNotiferProvider based on the currrentUser, which in return will notifyListeners() including a Widget, and in effect will auto-navigate whenever "something" happens to the firebaseAuth User?.

However, turns out, that events from the Stream<FirebaseUser?>, arrived when there was nothing to pop, hence the err.

Long story short, I believe I fixed it, by adding a canPop() before actually pop'ing the route.

Only reason for reporting this, was the " **In either case, please report this assertion by filing a bug on GitHub:** "
Should be good be OK to close the ticket.
